xesmf/smm.py

```diff
@@ -115,3 +116,34 @@ def apply_weights(weights, indata, shape_in, shape_out):
     outdata = outdata_flat.reshape(
         [*extra_shape, shape_out[0], shape_out[1]])
     return outdata
+
+
+def add_nans_to_weights(weights):
```

Comment:
@raphaeldussin, do we need a new test for this or is it being tested somewhere else?

Reply:
we should have tests for the whole module and end-to-end tests, maybe we can recycle the test problems from some of the notebooks

Reply:
function is now tested